Finished - Sprint 1, 2, 3, 4

IvanDok13 opened the pull request:
Sprint 1: Repository Setup

1. Task: [Repository Setup](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint%231.md)
2. Screenshot:
<img width="1440" alt="Снимок экрана 2025-05-02 в 19 57 23" src="https://github.com/user-attachments/assets/76c07d1f-4341-4c65-b2d6-6c020eff9071" />

3. Deploy: -
4. Done: 05.05.2025/06.05.2025
5. Evaluation Criteria 150/150

- [x] Repository Setup (29 points)
- [x] Task Board Setup (Trello, Jira, etc.) (14 points)
- [x] CommerceTools Project and API Client Setup (30 points)
- [x] Development Environment Configuration (43 points)
- [x] Development Scripts (15 points)
- [x] Comprehensive README (19 points)

Sprint 2: Login, Registration

1. Task: [Login, Registration](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint%232.md)
2. Screenshot:
<img width="1440" alt="Снимок экрана 2025-05-19 в 01 01 59" src="https://github.com/user-attachments/assets/45121510-22da-4ec7-9f08-d7865f6583be" />


3. Deploy: [eCommerce-ivanDok-Vladbes168-YuriyBiakov](https://ecommerce-ivandok-vladbes168-yuriybia.netlify.app/)
4. Done: 19.05.2025/19.05.2025
5. Evaluation Criteria 315/315

- [x] Code Review Participation (10 points)
- [x] No Merges Without Approvals (10 points)
- [x] Pull Request Descriptions (10 points)
- [x] Task Board Management (10 points)
- [x] Active Participation in Development Discussions (5 points)
- [x] Unit Test Coverage (15 points)

Login Page Implementation (130 points - Total)

1. Input Validation (40 points)
 - [x] (20 points) Implement client-side validation for the login form, including email and password fields.
 - [x] (20 points) Display clear error messages indicating any validation issues, such as an improperly formatted email.
2. Integration with Authentication Service (45 points)
 - [x] (25 points) Integrate the login form with a chosen authentication service (CommerceTools) to handle user authentication.
 - [x] (20 points) Implement error handling for failed authentication attempts, such as incorrect email or password, and display user-friendly error messages.
3. Redirection (30 points)
- [x] (15 points)  Redirect users to the application's main page upon successful login.
- [x] (15 points) Redirect users who are already logged in to the main page if they try to access the login page.
4. Handle Authentication Token (10 points)
- [x] (10 points) Obtain the authentication token securely after a successful login attempt by sending a request to the token endpoint (e.g., https://auth.europe-west1.gcp.commercetools.com/oauth/project_key/customers/token), allowing for seamless user authentication across the application.
5. Navigation to Registration Page (5 points)
 - [x] (5 points) Add a button or link on the login page that allows users to navigate to the registration page.
 
Registration Page Implementation (120 points - Total)

1. Input Validation (45 points)
 - [x] (25 points) Implement client-side validation for all required fields in the registration form, such as email, password, first name, last name, date of birth, and address fields (e.g., street, city, postal code, and country) for proper use with CommerceTools.
 - [x] (20 points) Display clear error messages indicating any validation issues, such as an improperly formatted email or a weak password.
2. Integration with Authentication Service (25 points)
 - [x] (10 points) Integrate the registration form with a chosen authentication service, such as commercetools, to handle user registration.
 - [x] (15 points) Implement error handling for failed registration attempts, and display user-friendly error messages.
3. State Management, Automatic Login, and Redirection (15 points)
 - [x] (15 points) Redirect users to the application's main page upon successful account creation and automatic login.
4. Integration with commercetools for User Profiles and Addresses (30 points)
 - [x] (15 points) Allow users to set a default address during registration.
 - [x] (15 points) Enable users to select different billing and shipping addresses or choose a single address for both billing and shipping during the registration process.
5. Navigation to Login Page (5 points)
 - [x] (5 points) Add a button or link on the registration page that allows users to navigate to the login page.
 
Main Page Enhancements (10 points - Total)

1. Centralized Navigation (10 points)
 - [x] (5 points) Add links to all the functional pages of the application on the main page. These should include, but are not limited to, the login and registration pages.
 - [x] (5 points) Each link should redirect the user correctly to the corresponding page without any errors.
2. Routing Implementation (30 points - Total)
 - [x] (15 points) Implement routing for navigation between login, registration, and main pages.
 - [x] (15 points) Implement a 404 (Not Found) page for invalid route requests.
3. Evaluation Criteria for Header (25 points - Total)
 - [x] (15 points) Navigation to login and registration pages for unauthorized users
 - [x] (5 points) Ability to access the main page for all users
 - [x] (5 points) Logout functionality for authorized users
 
 
 Sprint 3: Catalog Product Page, Detailed Product Page & User Profile Page Implementation
 
 1. Task: [Catalog Product Page, Detailed Product Page & User Profile Page Implementation](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint%233.md)
2. Screenshot:
<img width="1440" alt="Снимок экрана 2025-06-18 в 22 13 00" src="https://github.com/user-attachments/assets/d2fd8f7a-9c66-47c5-a3b3-a4865e8f73e6" />

3. Deploy: [eCommerce-ivanDok-Vladbes168-YuriyBiakov](https://ecommerce-ivandok-vladbes168-yuriybia.netlify.app/)
4. Done: 01.06.2025/01.06.2025
5. Evaluation Criteria 430/430
 
 Sprint 4: Basket Page, Catalog Page Enhancements, and About Us Page Implementation
 
 1. Task: [Basket Page, Catalog Page Enhancements, and About Us Page Implementation ](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint%234.md)
2. Screenshot:
<img width="1440" alt="Снимок экрана 2025-06-18 в 22 06 48" src="https://github.com/user-attachments/assets/a0775f18-3fab-4702-a08f-7cbc96ac980a" />

3. Deploy: [eCommerce-ivanDok-Vladbes168-YuriyBiakov](https://ecommerce-ivandok-vladbes168-yuriybia.netlify.app/)
4. Done: 16.06.2025/16.06.2025
5. Evaluation Criteria 345/345

Commit:
feat: add ct-config.ts file, env_example.txt file

Commit:
feat: add ct-config.ts file, env_example.txt file

Commit:
feat: pull updates from sprint-1/setup branch and keep single error type in ct-config.ts

Commit:
fix: conflicts

IvanDok13 committed:
Merge pull request #7 from IvanDok13/fix/1_21-change-pull-request-template

Fix/1_21 Change pull request template

IvanDok13 committed:
Merge pull request #4 from IvanDok13/feature/api-client

Feature/api client

IvanDok13 committed:
Merge pull request #8 from IvanDok13/sprint-1/setup

Sprint 1/setup

hhharm commented on package.json:
```diff
@@ -54,6 +55,7 @@
   "dependencies": {
     "@eslint/config-array": "^0.20.0",
```
all dependencies related to linting (esling, stylelint) should go to devDependencies - they are not needed in runtime

IvanDok13 replied:
Thank you for your feedback, we will fix this

YuriyBiakov committed:
feat: add DomElementAttribute and DomElementProperties interfaces

YuriyBiakov committed:
feat: implement createElementWithProperties function for dynamic element creation

IvanDok13 committed:
Merge pull request #10 from IvanDok13/feat/ECOMM-66_ECOMM-67-add-utils-and-types

Feat/ecomm 66 ecomm 67 add utils and types

IvanDok13 committed:
fix: move lint dependencies to devDependencies

IvanDok13 committed:
Merge pull request #11 from IvanDok13/fix/dependencies

fix/move lint dependencies to devDependencies

IvanDok13 committed:
feat: add html file

IvanDok13 committed:
feat: add react

IvanDok13 committed:
feat: update vite dependencies

IvanDok13 committed:
feat: update config whith react and short paths

IvanDok13 committed:
feat: update tsconfig with new alias for styles

IvanDok13 committed:
feat: update viteconfig for styles

IvanDok13 committed:
feat: update index.html

IvanDok13 committed:
feat: add logo

IvanDok13 committed:
feat: add default styles for project

IvanDok13 committed:
feat: update ts config whith path plugin

IvanDok13 committed:
feat: add router

IvanDok13 committed:
feat: add home page

IvanDok13 committed:
feat: add home styles

IvanDok13 committed:
feat: add main ts file

IvanDok13 committed:
feat: update home whith header

IvanDok13 committed:
feat: create header

IvanDok13 committed:
feat: add styles

IvanDok13 committed:
fix: remove types and utils

IvanDok13 committed:
fix: remove types

IvanDok13 committed:
fix: update types and selectors

IvanDok13 committed:
Merge pull request #13 from IvanDok13/feat/create-main-structure

Feat/create main structure

IvanDok13 committed:
Merge pull request #12 from IvanDok13/feat/update-setup

Feat/update setup

YuriyBiakov committed:
feat: add registration page component

YuriyBiakov committed:
feat: add CSS styles for registration page layout

Commit:
feat: create new component Login, add login_router, change links to header nav links

YuriyBiakov committed:
feat: add basic registration form layout with all required fields

YuriyBiakov committed:
feat: add CSS styles for registration form component

YuriyBiakov committed:
Merge pull request #14 from IvanDok13/feat/63-registration-form

Feat/63 registration form

YuriyBiakov committed:
feat: implement registration form in registration page

YuriyBiakov committed:
feat: add registration route and component to the router

IvanDok13 committed:
Merge pull request #15 from IvanDok13/feat/59_and_63-registration-page

Feat/59 and 63 registration page

YuriyBiakov committed:
refactor: move FormData type to a separate file for better organization

YuriyBiakov committed:
feat: add validation rules for form inputs

YuriyBiakov committed:
feat: add email input validation

YuriyBiakov committed:
style: enhance registration input styles for better layout

YuriyBiakov committed:
feat: add styles for error messages

YuriyBiakov committed:
feat: add validation functions for email, password and name

YuriyBiakov committed:
feat: implement validation functionality for remaining input fields

IvanDok13 committed:
Merge pull request #16 from IvanDok13/refactor/59_and_63-registration-page

refactor: move FormData type to a separate file for better organization

IvanDok13 committed:
Merge pull request #17 from IvanDok13/feat/68_validation-rules

Feat/68 validation rules

IvanDok13 committed:
Merge pull request #18 from IvanDok13/feat/40_and_41-reg_page-validation

Feat/40 and 41 reg page validation

Commit:
feat: create form page with all styles, components, types

Vladbes168 committed:
Merge branch 'sprint-2/setup' into feature/login-page

Commit:
fix: resolve conflicts, change env_example.txt

Commit:
Merge branch 'feature/login-page' of https://github.com/IvanDok13/eCommerce-Application into feature/login-page

Vladbes168 committed:
Merge pull request #20 from IvanDok13/feature/login-page

Feature/login page

YuriyBiakov committed:
chore: add commercetools packages

YuriyBiakov committed:
feat: add environment variable checks and constants

YuriyBiakov committed:
feat: implement CommerceTools client with environment variable validation

YuriyBiakov committed:
feat: add getProject function to fetch project data from CommerceTools API

YuriyBiakov committed:
chore: add debug button to fetch and display project information

YuriyBiakov committed:
feat: add registerCustomer function for customer registration

YuriyBiakov committed:
feat: implement customer registration form submission and mapping

YuriyBiakov committed:
feat: add loginCustomer function for user authentication

YuriyBiakov committed:
feat: enhance registration flow with navigation on successful registration

Vladbes168 committed:
Merge pull request #21 from IvanDok13/feat/69_ecommerce-api-setup

Feat/69 ecommerce api setup

Vladbes168 committed:
Merge pull request #22 from IvanDok13/feat/39-registration-api

Feat/39 registration api

YuriyBiakov committed:
feat: add toastify-js for enhanced toast notifications

YuriyBiakov committed:
chore: add TypeScript declaration for toastify-js module

YuriyBiakov committed:
feat: add showErrorToast utility for displaying error notifications

YuriyBiakov committed:
feat: integrate showErrorToast for error handling in registration form

YuriyBiakov committed:
chore: remove unused TODO comment

YuriyBiakov committed:
feat: enhance error handling in registration form with getErrorMessage utility

YuriyBiakov committed:
feat: improved error display when sending an incorrectly filled form

YuriyBiakov committed:
chore: delete test button for register page

YuriyBiakov committed:
feat: disabled state for registration button

Commit:
feat: add sign-up button that redirects to registration page

YuriyBiakov committed:
fix: format and clean up CSS for login form

YuriyBiakov committed:
refactor: rename class for registration button

YuriyBiakov committed:
refactor: change error display duration

YuriyBiakov committed:
refactor: change error display duration

Vladbes168 committed:
Merge pull request #23 from IvanDok13/feat/43-45_and_68-registration-page-functionality

Feat/43 45 and 68 registration page functionality

Commit:
feat: implement customer login API integration

Commit:
Merge branch 'sprint-2/setup' into feat/setup-api-ecommerce, pull data

YuriyBiakov committed:
feat: replace country input with dropdown and update validation rules

YuriyBiakov committed:
feat: add countries constant for dropdown selection

Commit:
feat: add login logic with Toastify

YuriyBiakov committed:
feat: restructure FormData to include shipping and billing addresses with optional fields

YuriyBiakov committed:
feat: enhance registration form to include separate shipping and billing addresses with country selection

YuriyBiakov committed:
feat: update registration form styles for address layout

YuriyBiakov committed:
style: change registration form container height to min-height for better responsiveness

YuriyBiakov committed:
feat: add options to set default shipping and billing addresses in registration form

YuriyBiakov committed:
style: rename CSS class for checkbox labels in registration form

YuriyBiakov committed:
style: refactor registration form layout with new container classes for improved structure and styling

YuriyBiakov committed:
feat: move default address checkbox labels to the bottom of their respective sections for better UX

IvanDok13 committed:
Merge pull request #24 from IvanDok13/feat/setup-api-ecommerce

Feat/setup api ecommerce

IvanDok13 committed:
Merge pull request #25 from IvanDok13/feat/45_and_68-address-setup-and-validation

Feat/45 and 68 address setup and validation

IvanDok13 committed:
Merge pull request #26 from IvanDok13/sprint-2/setup

Sprint 2/setup

Commit:
feat: enhance navigation system

- Add custom 404 Not Found page
- Implement useNavigate for back/home buttons
- Update router configuration

IvanDok13 committed:
Merge pull request #27 from IvanDok13/feat/404-page

feat: enhance navigation system

IvanDok13 committed:
feat: add jest

IvanDok13 committed:
feat: add tests for root

IvanDok13 committed:
feat: update config

IvanDok13 committed:
feat: add new packages

IvanDok13 committed:
fix: update error path

IvanDok13 committed:
feat: add tests for registration page

YuriyBiakov committed:
feat: update registration tests to use MemoryRouter for routing context

YuriyBiakov committed:
test: update password validation test to use async findByText

YuriyBiakov committed:
test: remove test for non-existent registration success message

YuriyBiakov committed:
test: update email validation test to use async findByText and find valid message

IvanDok13 committed:
Merge pull request #29 from IvanDok13/fix/unit-tests

Fix/unit tests

IvanDok13 committed:
Merge pull request #28 from IvanDok13/feat/unit-tests

Feat/unit tests

IvanDok13 committed:
refactor: remove unused export

IvanDok13 committed:
refactor: commit error

IvanDok13 committed:
refactor: add export

IvanDok13 committed:
feat: add footer

IvanDok13 committed:
feat: update font

IvanDok13 committed:
feat: update styles

IvanDok13 committed:
feat: update header

IvanDok13 committed:
refactor: change div tag for more descriptive main

IvanDok13 committed:
refactor: change div to main

IvanDok13 committed:
feat: update home page

IvanDok13 committed:
feat: update registration styles

IvanDok13 committed:
fix: remove styles

IvanDok13 committed:
Merge pull request #30 from IvanDok13/fix/remove-build-errors

Fix/remove build errors

IvanDok13 committed:
Merge pull request #31 from IvanDok13/feat/update-spa-layout

Feat/update spa layout

Commit:
fix: change validation for login form, add button visible password

Vladbes168 committed:
Merge pull request #35 from IvanDok13/fix/change-validation-rules

fix: change validation for login form, add button visible password

IvanDok13 committed:
fix: remove old api

IvanDok13 committed:
fix: rename files

IvanDok13 committed:
refactor: rename file

IvanDok13 committed:
fix: remove default settings for billing adress, rename file

IvanDok13 committed:
refactor: change file name and reg func

IvanDok13 committed:
feat: create user authRequestResponse func

IvanDok13 committed:
feat: create auth provider

IvanDok13 committed:
feat: update import

IvanDok13 committed:
feat: add token

IvanDok13 committed:
feat: add ecommerce keys

IvanDok13 committed:
fix: remove check func

IvanDok13 committed:
feat: add middleware options

IvanDok13 committed:
feat: add reg-user func

IvanDok13 committed:
feat: update add contextTypes

IvanDok13 committed:
feat: update styles

IvanDok13 committed:
fix: update component, add password and email check

IvanDok13 committed:
fix: remove additional styles

IvanDok13 committed:
feat: add error styles

IvanDok13 committed:
feat: add husky to project

IvanDok13 committed:
fix: rename file

IvanDok13 committed:
refactor: format styles

IvanDok13 committed:
refactor: format styles

IvanDok13 committed:
refactor: format styles

IvanDok13 committed:
feat: add path for api

IvanDok13 committed:
refactor: format variables

IvanDok13 committed:
refactor: format country constants

IvanDok13 committed:
refactor: format styles

IvanDok13 committed:
feat: update validation for trailspaces and whitespaces

IvanDok13 committed:
fix: change trail and white space validation

IvanDok13 committed:
feat: add error func

IvanDok13 committed:
fix: change validation rules

IvanDok13 committed:
feat: update setups

IvanDok13 committed:
feat: remove toastyfy

YuriyBiakov committed:
fix: change imports and functions return types

YuriyBiakov committed:
fix: update imports to avoid errors

YuriyBiakov committed:
fix: change imports and functions return types

YuriyBiakov committed:
fix: change function name to avoid error

YuriyBiakov committed:
fix: change function return type  to avoid using any

YuriyBiakov committed:
chore: add temp lines to ensure correct build during refactoring

IvanDok13 committed:
Merge pull request #37 from IvanDok13/review-fix/login-auth-funcs

Review fix/login auth funcs

YuriyBiakov committed:
Merge pull request #36 from IvanDok13/fix/login-auth-funcs

Fix/update login and auth funcs with user token

IvanDok13 committed:
feat: add styles for show-passw btn

IvanDok13 committed:
fix: brought to a unified style

IvanDok13 committed:
fix: brought to a unified style

IvanDok13 committed:
fix: brought to a unified style

IvanDok13 committed:
fix: brought to a unified style

IvanDok13 committed:
refactor: change validation text

IvanDok13 committed:
refactor: move common styles

IvanDok13 committed:
fix: change validation rules for street

IvanDok13 committed:
refactor: move common styles

IvanDok13 committed:
fix: remove unnecessary styles

IvanDok13 committed:
feat: add env

IvanDok13 committed:
feat: update gitignore

IvanDok13 committed:
feat: update validation

IvanDok13 committed:
feat: update validation and styles for reg form

YuriyBiakov committed:
Merge pull request #38 from IvanDok13/fix/login-reg-make-same-format

Fix/login reg make same format

IvanDok13 committed:
Merge pull request #33 from IvanDok13/sprint-2/setup

Sprint 2/setup
